### PR TITLE
Remove usage of np.int, np.float and np.bool

### PR DESCRIPTION
**Pull request by burk**
They are aliases for the built-in types and are deprecated in newer numpy versions.

### CODE REVIEW

**Review by simonangerbauer — APPROVED** [review]

File: tests/explainers/test_tree.py [hunk 1]
```diff
@@ -391,8 +391,13 @@ def test_gpboost():
 def test_catboost():
     catboost = pytest.importorskip("catboost")
     # train catboost model
+<<<<<<< HEAD
+    X, y = shap.datasets.boston()
+    X["RAD"] = X["RAD"].astype(int)
+=======
     X, y = shap.datasets.california(n_points=500)
     X['IsOld'] = (X['HouseAge'] > 30).astype(str)
+>>>>>>> 45b85c1837283fdaeed7440ec6365a886af4a333
```
Comment (simonangerbauer) [inline tests/explainers/test_tree.py]: Thanks for updating the PR.
There is still an open merge conflict here.

**Review by simonangerbauer — CHANGES_REQUESTED** [review]

File: tests/explainers/test_tree.py (hunk 1 above)
Comment (burk) [reply tests/explainers/test_tree.py]: Oops, thanks!

**Comment by burk** [comment]
I also saw `np.integer` was used in some `isinstance` checks, but I'm not sure if that is deprecated.

File: tests/explainers/test_tree.py (hunk 1 above)
Comment (mrzdev) [reply tests/explainers/test_tree.py]: Found an unnecessary dot here
https://github.com/burk/shap/commit/be2658e62df109e0d66b448652d7f0b8cb05619f

**Comment by paapu88** [comment]
I have pip installed
shap==0.41.0
and 
numpy==1.24.3
getting error
```
In the future `np.bool` will be defined as the corresponding NumPy scalar.
Traceback (most recent call last):
  File "/home/hu-mka/git/skyfora-seasonal/seasonal/scripts/prediction/update.py", line 89, in <module>
    preds, shap_vals = seasonal_prediction(data_x, data_y, model, categories_hierarchy, model_args,
  File "/home/hu-mka/git/skyfora-seasonal/seasonal/utils/model.py", line 101, in seasonal_prediction
    explainer = shap.Explainer(shap_predict, train_x)
  File "/home/hu-mka/venvs/seasonal/lib/python3.10/site-packages/shap/explainers/_explainer.py", line 88, in __init__
    self.masker = maskers.Independent(masker)
  File "/home/hu-mka/venvs/seasonal/lib/python3.10/site-packages/shap/maskers/_tabular.py", line 262, in __init__
    super().__init__(data, max_samples=max_samples, clustering=None)
  File "/home/hu-mka/venvs/seasonal/lib/python3.10/site-packages/shap/maskers/_tabular.py", line 82, in __init__
    self._last_mask = np.zeros(data.shape[1], dtype=np.bool)
  File "/home/hu-mka/venvs/seasonal/lib/python3.10/site-packages/numpy/__init__.py", line 305, in __getattr__
    raise AttributeError(__former_attrs__[attr])
AttributeError: module 'numpy' has no attribute 'bool'.
`np.bool` was a deprecated alias for the builtin `bool`. To avoid this error in existing code, use `bool` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.bool_` here.
The aliases was originally deprecated in NumPy 1.20; for more details and guidance see the original release note at:
    https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations. Did you mean: 'bool_'?

```

**Comment by zhimin-z** [comment]
I have met the exact problem using shap:
```
AttributeError: module 'numpy' has no attribute 'int'.
`np.int` was a deprecated alias for the builtin `int`. To avoid this error in existing code, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
The aliases was originally deprecated in NumPy 1.20; for more details and guidance see the original release note at:
    https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
```
Has the merged part been released to the pypi package yet? I am using `0.41.0` but still get this error.

**Comment by burk** [comment]
This PR has not been merged, and I don't really expect it to be, given the low activity on this repo. But dsgibbons has merged the fix into https://github.com/dsgibbons/shap. Maybe they intend on pushing an alternative pypi package.

**Comment by detrin** [comment]
> This PR has not been merged, and I don't really expect it to be, given the low activity on this repo. But dsgibbons has merged the fix into https://github.com/dsgibbons/shap. Maybe they intend on pushing an alternative pypi package.

This sucks tbh because I rely heavily on this package and I can't stick to numpy < 1.24. Thanks for the reference

**Comment by burk** [comment]
See also https://github.com/dsgibbons/shap/discussions/1. They're looking for maintainers to be able to fix existing issues and build an alternative package.

**Comment by zhimin-z** [comment]
It loos like this has already been been merged. Refer to https://github.com/dsgibbons/shap/pull/7
When will it be released in pypi then?

**Comment by arareko** [comment]
> It loos like this has already been been merged. Refer to [dsgibbons#7](https://github.com/dsgibbons/shap/pull/7) When will it be released in pypi then?

@zhimin-z Merged, but in a different user/repository. Whatever that builds must be on a different package on PyPI.

**Comment by arareko** [comment]
There's a PR that has been opened for a while to fix the Numpy deprecations https://github.com/slundberg/shap/pull/2048

**Comment by sergylog** [comment]
Is this issue going to be solved?

**Comment by thatlittleboy** [comment]
Duplicated by #2943

**Pull request closed without merging**